viewer/src/graphics/utils.ts

```diff
@@ -25,6 +26,54 @@ export function planetClass(faction: string): string {
     case "automa":
       return "gen";
     default:
-      return factions.planet(faction as Faction);
+      return factionPlanet(faction as Faction);
   }
 }
+
+export function lightenDarkenColor(col: string, amt: number) {
+  let usePound = false;
+
+  if (col[0] == "#") {
+    col = col.slice(1);
+    usePound = true;
+  }
+
+  const num = parseInt(col, 16);
+
+  let r = (num >> 16) + amt;
+
+  if (r > 255) r = 255;
+  else if (r < 0) r = 0;
+
+  let b = ((num >> 8) & 0x00ff) + amt;
+
+  if (b > 255) b = 255;
+  else if (b < 0) b = 0;
+
+  let g = (num & 0x0000ff) + amt;
+
+  if (g > 255) g = 255;
+  else if (g < 0) g = 0;
+
+  return (usePound ? "#" : "") + (g | (b << 8) | (r << 16)).toString(16);
+}
+
+function newPlanetColors(amt: number) {
+  return Object.fromEntries(
+    Object.entries(factions).map(([f, c]) => {
+      const planet = c.planet;
+      const color = planet == Planet.Ice ? "#000000" : planets[planet].color;
+      return [f, amt == 0 ? color : lightenDarkenColor(color, amt)];
+    })
+  );
+}
+
+export const factionLogTextColors = Object.fromEntries(
+  Object.entries(factions).map(([f, c]) => {
+    const planet = c.planet;
+    const color = planet == Planet.Ice || planet == Planet.Swamp || planet == Planet.Titanium ? "white" : "black";
+    return [f, color];
+  })
+);
+export const factionLogColors = newPlanetColors(0);
+export const lightFactionLogColors = newPlanetColors(190);
```

Comment:
I think all this can be avoided by doing something like:

```
.advanced-log {
  /* darken */
  .ivits: { color: var(--oxide); filter: brightness(50%)}
  /* or lighten */
  .ivits: { color: var(--oxide); filter: brightness(150%)}
}
```

Or maybe introduce scss variables based on factions, and use those instead of the planets. It's less added code and doesn't need to run javascript. Easy to change an individual settings as well directly from the css file.

Reply:
problem is that the brightness also applies to the text

![image](https://user-images.githubusercontent.com/2832627/114352513-3ac9a980-9b6c-11eb-8d93-40dc0582ec33.png)

Reply:
Too bad css doesn't have a darken function like scss, and scss's darken function only works on scss variables and not css ones :(